src/Microsoft.ComponentDetection.Detectors/rust/RustCliDetector.cs

```diff
@@ -118,22 +118,28 @@ protected override async Task OnFileFoundAsync(ProcessRequest processRequest, ID
                             string.IsNullOrWhiteSpace(x.License) ? null : x.License));
 
                     var root = metadata.Resolve.Root;
+                    HashSet<string> visitedDependencies = new();
 
                     // A cargo.toml can be used to declare a workspace and not a package (A Virtual Manifest).
                     // In this case, the root will be null as it will not be pulling in dependencies itself.
                     // https://doc.rust-lang.org/cargo/reference/workspaces.html#virtual-workspace
                     if (root == null)
                     {
-                        this.Logger.LogWarning("Virtual Manifest: {Location}, falling back to cargo.lock parsing", processRequest.ComponentStream.Location);
-                        record.DidRustCliCommandFail = true;
-                        record.WasRustFallbackStrategyUsed = true;
-                        record.FallbackReason = "Virtual Manifest";
-                    }
+                        this.Logger.LogWarning("Virtual Manifest: {Location}", processRequest.ComponentStream.Location);
 
-                    HashSet<string> visitedDependencies = new();
-                    if (!record.WasRustFallbackStrategyUsed)
+                        foreach (var dep in metadata.Resolve.Nodes)
+                        {
+                            var componentKey = $"{dep.Id}";
+                            if (!visitedDependencies.Contains(componentKey))
```

Comment:
Can we add some unit tests to validate this new logic?

Reply:
added